ostree-pulp.sh

```diff
@@ -0,0 +1,405 @@
+#!/bin/bash
+set -euox pipefail
+
+# Provision the software under test.
+./setup.sh
+
+# Get OS data.
+source /etc/os-release
+ARCH=$(uname -m)
+
+# Set up variables.
+TEST_UUID=$(uuidgen)
+IMAGE_KEY="ostree-raw-${TEST_UUID}"
+PROD_REPO_URL=http://192.168.100.1/repo
+PROD_REPO=/var/www/html/repo
+GUEST_ADDRESS=192.168.100.50
+
+# Set up temporary files.
+TEMPDIR=$(mktemp -d)
+BLUEPRINT_FILE=${TEMPDIR}/blueprint.toml
+KS_FILE=${TEMPDIR}/ks.cfg
+COMPOSE_START=${TEMPDIR}/compose-start-${IMAGE_KEY}.json
+COMPOSE_INFO=${TEMPDIR}/compose-info-${IMAGE_KEY}.json
+
+# SSH setup.
+SSH_OPTIONS=(-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o ConnectTimeout=5)
+SSH_KEY=key/ostree_key
+SSH_KEY_PUB=$(cat "${SSH_KEY}".pub)
+
+# Prepare osbuild-composer repository file
+sudo mkdir -p /etc/osbuild-composer/repositories
+
+# RHEL 8.8 and CS8 is still RO for /sysroot on raw image and simplified installer
+# The RO setting on RHEL 8.8 and CS8 is not configured by ostree, but osbuild-composer
+# by PR https://github.com/osbuild/osbuild-composer/pull/3178
+case "${ID}-${VERSION_ID}" in
+    "rhel-9.3")
+        IMAGE_TYPE=edge-commit
+        OSTREE_REF="rhel/9/${ARCH}/edge"
+        OS_VARIANT="rhel9-unknown"
+        BOOT_LOCATION="http://${DOWNLOAD_NODE}/rhel-9/nightly/updates/RHEL-9/latest-RHEL-9.3.0/compose/BaseOS/x86_64/os/"
+        ;;
+    *)
+        echo "unsupported distro: ${ID}-${VERSION_ID}"
+        exit 1;;
+esac
+
+# Pulp setup
+PULP_CONFIG_FILE="$(pwd)/pulp.toml"
```

Comment:
Please put temp file into TEMPDIR, not current folder. Or please remove it if test finished.

Reply:
now these files will be removed in cleanup step.